packages/core/src/main/notification/NotificationService.ts

```diff
@@ -519,4 +521,62 @@ export class NotificationService extends EventEmitter {
       this.logger.error('Could not get last key material update dates', error);
     }
   }
+
+  private scheduleTaskToQueryKeyPackagesCountAndSyncWithBackend(firingDate: number) {
+    TaskScheduler.addTask({
+      firingDate,
+      key: 'try-key-packages-backend-sync',
+      task: () => this.queryKeyPackagesCountAndSyncWithBackend(),
+    });
+  }
+
+  private async queryKeyPackagesCountAndSyncWithBackend() {
+    const validKeyPackagesCount = await this.mlsService.clientValidKeypackagesCount();
+
+    const lastQueryDate = new Date().getTime();
+
+    await this.database.storeLastKeyPackageQueryDate({lastQueryDate});
+
+    const minAllowedNumberOfKeyPackages = INITIAL_NUMBER_OF_KEY_PACKAGES / 2;
+
+    if (validKeyPackagesCount <= minAllowedNumberOfKeyPackages) {
+      const clientId = this.apiClient.validatedClientId;
+
+      //check numbers of keys on backend
+      const backendKeyPackagesCount = await this.apiClient.api.client.getMLSKeyPackageCount(clientId);
+
+      if (backendKeyPackagesCount <= minAllowedNumberOfKeyPackages) {
+        //upload new keys
+        const newKeyPackages = await this.mlsService.clientKeypackages(INITIAL_NUMBER_OF_KEY_PACKAGES);
+
+        await this.mlsService.uploadMLSKeyPackages(newKeyPackages, clientId);
```

Comment:
❓ Is uploading the keys all we have to do? Don't we want to update the keys locally also, so we can use new keys right away in current client session? 🤔 Unless it's somehow done automatically under the hood 🤔

Reply:
do you need the keys locally? i think keys always just live on the backend and we call the backend to get them as needed

Reply:
hmmm, the task is about syncing, so my thought is it should sync the packages on both (client and backend) sides 🤔

Reply:
Humm yeah I think there is something wrong here 🤔 
Normally it should be coreCrypto that generates (and store) the new key material. There must be a method for this. 
And you should be requesting `INITIAL_NUMBER.. - validKeyPackageCount` I believe (we just want to generate the difference, not add another 100 to it, I believe)

Reply:
My understanding was that they handle it on `core-cypto` side.

<img width="766" alt="Screenshot 2022-09-12 at 16 57 40" src="https://user-images.githubusercontent.com/45733298/189687557-a650048f-bf7f-415d-801b-00b356d3599d.png">

Reply:
ohhh thanks for linking the doc, it then makes perfect sense 👌